src/odemis/driver/xt_client.py

```diff
@@ -1915,6 +1980,11 @@ def __init__(self, name, role, parent, **kwargs):
         self.position = model.VigilantAttribute({}, unit="m", readonly=True)
         self._updatePosition()
 
+        if not hasattr(self.parent, "_scanner"):
+            if not hasattr(self.parent, "_mb-scanner"):
```

Comment:
There is no such `"_mb-scanner"` (or `"_mb_scanner"`). It's always called `_scanner`. So, you can drop this if.